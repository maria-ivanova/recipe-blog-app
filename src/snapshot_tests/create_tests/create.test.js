import React from 'react';
import renderer from 'react-test-renderer';
import Routing from '../helpers/router.js';
import Create from '../../components/create.js';

import { library } from '@fortawesome/fontawesome-svg-core';
import { faUpload } from '@fortawesome/free-solid-svg-icons';
library.add(faUpload);

jest.mock('../../components/notifications.js', () => 'Notifications');

describe('Create Component', () => {
    it('should render page layout', () => {
        const tree = renderer.create(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "vMmkqnOVIzQ2PjtoBZ9IkogAtt22" }}>
                <Create />
            </Routing>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})