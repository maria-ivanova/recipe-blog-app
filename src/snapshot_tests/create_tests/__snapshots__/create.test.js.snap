// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Component should render page layout 1`] = `
<section
  className="sec content_sec"
>
  <div
    className="container tcenter"
  >
    <Notifications />
    <h2
      className="page_title"
    >
      <span>
        Създай рецепта
      </span>
    </h2>
    <form
      className="recipe_form"
      onSubmit={[Function]}
    >
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Заглавие 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <input
          className="input_text input_text"
          name="title"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Категория 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <select
          className="input_text input_text"
          name="category"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Време за приготвяне (макс. 10 символа) 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <input
          className="input_text input_text"
          name="totalTime"
          onChange={[Function]}
          placeholder="20 мин. / 1 час / 2 часа"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Порции 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <input
          className="input_text input_text"
          name="yields"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Качване на снимка 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <span
          className="img_holder"
        >
          Все още няма качена снимка
        </span>
        <label
          className="btn upload_btn"
        >
          <input
            onChange={[Function]}
            type="file"
          />
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-upload fa-w-16 fa"
            data-icon="upload"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          качи
        </label>
      </div>
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Необходими съставки 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <textarea
          className="input_text input_text"
          name="ingredients"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <div
        className="input_row"
      >
        <label
          className="lbl"
        >
          Начин на приготвяне 
          <span
            className="tred"
          >
            *
          </span>
        </label>
        <textarea
          className="input_text input_text"
          name="recipeDescription"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <div
        className="input_row"
      >
        <button
          className="btn"
          type="submit"
        >
          Създай
        </button>
      </div>
    </form>
  </div>
</section>
`;
